htdocs/core/class/commondocgenerator.class.php

```diff
@@ -408,8 +408,8 @@ public function get_substitutionarray_object($object, $outputlangs, $array_key =
 			$sumpayed = $object->getSommePaiement();
 			$sumdeposit = $object->getSumDepositsUsed();
 			$sumcreditnote = $object->getSumCreditNotesUsed();
-			$already_payed_all = $sumpayed + $sumdeposit + $sumcreditnote;
-			$remain_to_pay = $sumpayed - $sumdeposit - $sumcreditnote;
+			$already_payed_all = $sumpayed - $sumdeposit - $sumcreditnote;
```

Comment:
The already_payed_all should be the sum of all. So old code seems good.
Can you reproduce the bug on demo.dolibarr.org ?

Reply:
Hello

Behavior is different between my instance and demo.
In my demo "remise" is set a standard payment but in demo is set as a negative service

Demo : new service row with negative amount
![Capture d’écran de 2021-11-08 09-36-06](https://user-images.githubusercontent.com/209915/140709853-f7896fc1-0f1d-40a3-9ce6-cd3e4ff01986.png)

Prod : new payment line to report acompte invoice
![2021-11-08_09-36](https://user-images.githubusercontent.com/209915/140709883-3e8395e7-1065-46b2-ad87-c68a148a955b.png)

Reply:
You probably enabled the option FACTURE_DEPOSITS_ARE_JUST_PAYMENTS ?
This option is an old an bugged implementation that should not be used. There is ton of bug with it and it is compliant with legal rules of accountancy.